packages/astro/src/actions/runtime/middleware.ts

```diff
@@ -47,7 +47,15 @@ export const onRequest = defineMiddleware(async (context, next) => {
 		},
 	};
 	Object.defineProperty(locals, '_actionsInternal', { writable: false, value: actionsInternal });
-	return next();
+	const response = await next();
+	if (result.error) {
+		return new Response(response.body, {
+			status: result.error.status,
+			statusText: result.error.message,
+			headers: response.headers,
+		})
```

Comment:
Maybe we should return a simplified `statusText` for prod? In https://github.com/withastro/astro/pull/10284, it was considered a vulnerability that the error message was shown to the users.

Reply:
Good point. I also realized `statusText` does not support special characters like newlines. To keep things simple, I'll just out the `error.name` here, which is just the status code written out like `NOT_FOUND`. No need for prod security checks I don't think